Makefile

```diff
@@ -1,8 +1,12 @@
-.PHONY: all build
+.PHONY: all build lint
 all: build
 
 build:
 	go build -o bin/kubectl-operator
 
 install: build
 	install bin/kubectl-operator $(shell go env GOPATH)/bin
+
+lint:
+	scripts/golangci-lint-check.sh
+	golangci-lint --timeout 3m run
```

Comment:
Instead of adding a separate script, wdyt about putting the script into a `golangci-lint` make target kind of like [this](https://github.com/operator-framework/operator-sdk/blob/f04ad0730a655ec0f3aee06a573e3b74205b44af/internal/plugins/helm/v1/scaffolds/internal/templates/makefile.go#L111-L121), except that it would check presence/version in `./bin`, download the right version of `golangci-lint` if necessary into `./bin`, and then always uses it from there.

Developers might be working on different projects that download tools in `$GOBIN`, so this approach would avoid different projects fighting with each other if they use different versions of the same tool.

Reply:
I can switch to that... I would prefer to run from our bin actually.
I try to stay away from script functions in my make file however... it is less portable... among other things... but this makes sense for this.   I will have it tomorrow.